contracts/finance/andromeda-validator-staking/src/contract.rs

```diff
@@ -166,8 +168,12 @@ fn execute_unstake(
         });
     };
 
+    let unstake_amount = amount.unwrap_or(res.amount.amount);
+
     ensure!(
-        !res.amount.amount.is_zero(),
+        !res.amount.amount.is_zero()
```

Comment:
I think `!res.amount.amount.is_zero()` is unnecessary.
```
            !unstake_amount.is_zero()
            && unstake_amount <= res.amount.amount,
```
We already ensure that `unstake_amount` is non zero and `res.amount.amount` is not less than `unstake_amount`